tests/core/test_evaluation.py

```diff
@@ -242,3 +244,38 @@ def test_event_has_proper_implementation(
     actual_entities = _clean_entity_results(text, [entity])
 
     assert actual_entities[0] == expected_entity
+
+
+async def test_retrieval_intent(response_selector_agent: Agent):
+    generator = await _create_data_generator(
+        RETRIEVAL_INTENT_STORY_FILE, response_selector_agent, use_e2e=True
+    )
+    test_stories = generator.generate_story_trackers()
+
+    story_evaluation, num_stories = await _collect_story_predictions(
+        test_stories, response_selector_agent, use_e2e=True
+    )
+    # check that test story can either specify base intent or full retrieval intent
+    assert not story_evaluation.evaluation_store.has_prediction_target_mismatch()
```

Comment:
I think it would be better if these would be two test cases. one testing with base intent and the other with the full retrieval intent